components/flow-orchestration-framework/org.wso2.carbon.identity.flow.mgt/src/main/java/org/wso2/carbon/identity/flow/mgt/dao/CacheBackedFlowDAOImpl.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.flow.mgt.dao;
+
+import org.wso2.carbon.identity.flow.mgt.cache.FlowMgtCache;
+import org.wso2.carbon.identity.flow.mgt.cache.FlowMgtCacheKey;
+import org.wso2.carbon.identity.flow.mgt.cache.GraphConfigCache;
+import org.wso2.carbon.identity.flow.mgt.exception.FlowMgtFrameworkException;
+import org.wso2.carbon.identity.flow.mgt.exception.FlowMgtServerException;
+import org.wso2.carbon.identity.flow.mgt.model.FlowDTO;
+import org.wso2.carbon.identity.flow.mgt.model.GraphConfig;
+import org.wso2.carbon.identity.flow.mgt.utils.FlowMgtUtils;
+
+import static org.wso2.carbon.identity.flow.mgt.Constants.DEFAULT_FLOW_NAME;
+
+/**
+ * Cache backed implementation of {@link FlowDAO}.
+ * This class provides caching capabilities for flow management operations.
+ */
+public class CacheBackedFlowDAOImpl implements FlowDAO {
+
+    private static final CacheBackedFlowDAOImpl INSTANCE = new CacheBackedFlowDAOImpl();
+    private static final FlowDAOImpl FLOW_DAO = new FlowDAOImpl();
+
+    private CacheBackedFlowDAOImpl() {
+
+    }
+
+    public static CacheBackedFlowDAOImpl getInstance() {
+
+        return INSTANCE;
+    }
+
+    @Override
+    public void updateFlow(String flowType, GraphConfig graphConfig, int tenantId, String flowName)
+            throws FlowMgtFrameworkException {
+
+        FlowMgtCacheKey flowMgtCacheKey = new FlowMgtCacheKey(flowType, flowName);
+        FlowMgtUtils.clearCache(flowMgtCacheKey, FlowMgtCache.getInstance(), tenantId);
+        FlowMgtUtils.clearCache(flowMgtCacheKey, GraphConfigCache.getInstance(), tenantId);
```

Comment:
Here, updating flow of one tenant clears the FlowMgtCache and GraphConfigCache of all sub orgs FlowMgtCache and GraphConfigCache as well. 
Why it's required?
If sub org has a value for FlowMgtCache or GraphConfigCache means they have their own overriden flow.

Isn't that we just need to clear resolver cache of sub orgs, which is handled in service layer?

Reply:
Fixed